genie-web/src/main/java/com/netflix/genie/web/events/AbstractSNSPublisher.java

```diff
@@ -37,81 +35,46 @@
 import java.util.UUID;
 
 /**
- * Publishes Amazon SNS notifications.
- * Currently, only job status changes trigger a notification.
+ * Abstract SNS publisher that adds fields common to all payloads, records metrics, etc.
  *
  * @author mprimi
  * @since 4.0.0
  */
 @Slf4j
-public class SNSNotificationsPublisher implements ApplicationListener<JobStateChangeEvent> {
-
+public class AbstractSNSPublisher {
```

Comment:
should you make this abstract and implement `ApplicationListener<JobStateChangedEvent>` or is there a thought that maybe others could publish off different internal events?

Reply:
I should make this class abstract. But I'd rather not make it specific to `JobStateChangedEvent`

Reply:
Ok then you could make it `implements ApplicationListener<? extends ApplicationEvent>`